mobile/src/containers/Settings.jsx

```diff
@@ -31,8 +31,17 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
     dispatch(unlink(client))
     ownProps.router.replace('/onboarding')
   },
-  setBackupImages: (e) => dispatch(setBackupImages(e.target.checked)),
-  setWifiOnly: (e) => dispatch(setWifiOnly(e.target.checked)),
+  setBackupImages: async (e) => {
+    const value = e.target.checked
+    const action = await dispatch(backupImages(value))
+    if (value === true && value !== action) {
+      e.target.checked = false
```

Comment:
@kosssi not good, if the checkbox state is controlled by a reducer, it must stay that way: manipulating DOM directly like this is an antipattern. Your `backupImages` action creator should dispatch whatever actions necessary to update the reducer for the checkbox state.